src/history/__tests__/historyActions.test.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  CHANGE_DATE_FILTER_ACTION,
+  CHANGE_PRICE_FILTER_ACTION,
+  changeDateFilterAction,
+  changePriceFilterAction,
+  getHistoryByDateFilter,
+  getHistoryByPriceFilter,
+  getHistoryData,
+  HISTORY_RECEIVED_ACTION,
+  historyReceivedAction
+} from 'history/historyActions'
+import configureMockStore from 'redux-mock-store'
+import thunk from 'redux-thunk'
+
+import mockApi from './__mock__/mockApi'
+import { mockChartData, mockEmptyHistoryState, mockHistoryState } from './__mock__/mockData'
+
+const setup = () => {
+  const middlewares = [thunk.withExtraArgument(mockApi)]
+  return configureMockStore(middlewares)
+}
+
+describe('history actions', () => {
+  it('should create action update history data', () => {
+    const expectedAction = {
+      type: HISTORY_RECEIVED_ACTION,
+      history: mockChartData
+    }
+    expect(historyReceivedAction(mockChartData)).toEqual(expectedAction)
+  })
+
+  it('should create OVERVIEW_RECEIVED_ACTION after successfully fetching history', () => {
+    const mockStore = setup()
+    const store = mockStore({ history: mockEmptyHistoryState })
+
+    const expectedAction = [
+      {
+        type: HISTORY_RECEIVED_ACTION,
+        history: mockHistoryState
+      }
+    ]
+    return store.dispatch(getHistoryData('AAPL') as any).then(() => {
```

Comment:
you could make this into an async function and then `await store.dispatch(getHistoryData('AAPL'))`